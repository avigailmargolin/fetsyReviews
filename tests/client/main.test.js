import React from 'react';
import renderer from 'react-test-renderer';
import sampleReviews from './sampleReview';
import ReviewsNav from '../../client/src/components/reviewsNav.jsx';
import CustomerPhotos from '../../client/src/components/customerPhotos.jsx';
import SingleReview from '../../client/src/components/singleReview.jsx';
import TotalReview from '../../client/src/components/totalReview.jsx';
import Main from '../../client/src/components/main.jsx';
import ListingPageReviews from '../../client/src/components/listingPageReviews.jsx';
import DataReviews from '../../client/src/components/dataReviews.jsx';

describe('Component snapshot renders correctly inital layout', () => {
	test('listing page reviews renders', () => {
		const component = renderer.create(<ListingPageReviews reviews={sampleReviews.sampleReviews} />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	test('data reviews renders', () => {
		const component = renderer.create(<DataReviews reviews={sampleReviews.sampleReviews} />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	test('reviews nav renders', () => {
		const component = renderer.create(<ReviewsNav reviews={sampleReviews.sampleReviews} />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	test('customer photos renders', () => {
		const component = renderer.create(<CustomerPhotos reviews={sampleReviews.sampleReviews} />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	test('single review renders', () => {
		const component = renderer.create(<SingleReview review={sampleReviews.sampleReviews[0]} />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	test('total review renders', () => {
		const component = renderer.create(<TotalReview reviews={sampleReviews.sampleReviews} />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
