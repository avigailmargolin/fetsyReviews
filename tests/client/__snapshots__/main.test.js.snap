// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshot renders correctly inital layout customer photos renders 1`] = `
<div>
  <img
    src="https://source.unsplash.com/random/200x200"
  />
  <img
    src="https://source.unsplash.com/random/200x200"
  />
  <img
    src="https://source.unsplash.com/random/200x200"
  />
  <img
    src="https://source.unsplash.com/random/200x200"
  />
  <img
    src="https://source.unsplash.com/random/200x200"
  />
  <img
    src="https://source.unsplash.com/random/200x200"
  />
</div>
`;

exports[`Component snapshot renders correctly inital layout data reviews renders 1`] = `
<div>
  <div>
    Reviews for this item 
    6
     
    <br />
    Review for this shop 1,088
    <div>
      Sort by: Recommended
    </div>
  </div>
  <div>
    <div
      style={
        Object {
          "border": "2px solid black",
        }
      }
    >
      Kazza
       
      <br />
      Nov 11, 2020
       
      <br />
      5
       star 
      <br />
      Mask color: 
      Christmas tree+bow
       
      <br />
      Speedy delivery Item nice Recommend
       
      <br />
    </div>
    <div
      style={
        Object {
          "border": "2px solid black",
        }
      }
    >
      Angela Naylor
       
      <br />
      Oct 21, 2020
       
      <br />
      5
       star 
      <br />
      Mask color: 
      Christmas tree+bow
       
      <br />
      Absolutely beautiful will definitely be purchasing some more . Delivery time outstanding so quick thank you so much xx ðŸ’œ
       
      <br />
    </div>
    <div
      style={
        Object {
          "border": "2px solid black",
        }
      }
    >
      ruthiebaby1619
       
      <br />
      Nov 9, 2020
       
      <br />
      5
       star 
      <br />
      Mask color: 
      Christmas tree+bow
       
      <br />
      This mask is fabulous! Glitter design at both sides. Very comfortable. Can't wait till December to wear it!
       
      <br />
    </div>
    <div
      style={
        Object {
          "border": "2px solid black",
        }
      }
    >
      Kazza
       
      <br />
      Nov 11, 2020
       
      <br />
      5
       star 
      <br />
      Mask color: 
      Christmas tree+bow
       
      <br />
      Speedy delivery Item nice Recommend
       
      <br />
    </div>
    <div
      style={
        Object {
          "border": "2px solid black",
        }
      }
    >
      Angela Naylor
       
      <br />
      Oct 21, 2020
       
      <br />
      5
       star 
      <br />
      Mask color: 
      Christmas tree+bow
       
      <br />
      Absolutely beautiful will definitely be purchasing some more . Delivery time outstanding so quick thank you so much xx ðŸ’œ
       
      <br />
    </div>
    <div
      style={
        Object {
          "border": "2px solid black",
        }
      }
    >
      ruthiebaby1619
       
      <br />
      Nov 9, 2020
       
      <br />
      5
       star 
      <br />
      Mask color: 
      Christmas tree+bow
       
      <br />
      This mask is fabulous! Glitter design at both sides. Very comfortable. Can't wait till December to wear it!
       
      <br />
    </div>
  </div>
</div>
`;

exports[`Component snapshot renders correctly inital layout listing page reviews renders 1`] = `
<div>
  <div>
    6
     reviews 
    5
     rating stars
  </div>
  <div>
    <div>
      Reviews for this item 
      6
       
      <br />
      Review for this shop 1,088
      <div>
        Sort by: Recommended
      </div>
    </div>
    <div>
      <div
        style={
          Object {
            "border": "2px solid black",
          }
        }
      >
        Kazza
         
        <br />
        Nov 11, 2020
         
        <br />
        5
         star 
        <br />
        Mask color: 
        Christmas tree+bow
         
        <br />
        Speedy delivery Item nice Recommend
         
        <br />
      </div>
      <div
        style={
          Object {
            "border": "2px solid black",
          }
        }
      >
        Angela Naylor
         
        <br />
        Oct 21, 2020
         
        <br />
        5
         star 
        <br />
        Mask color: 
        Christmas tree+bow
         
        <br />
        Absolutely beautiful will definitely be purchasing some more . Delivery time outstanding so quick thank you so much xx ðŸ’œ
         
        <br />
      </div>
      <div
        style={
          Object {
            "border": "2px solid black",
          }
        }
      >
        ruthiebaby1619
         
        <br />
        Nov 9, 2020
         
        <br />
        5
         star 
        <br />
        Mask color: 
        Christmas tree+bow
         
        <br />
        This mask is fabulous! Glitter design at both sides. Very comfortable. Can't wait till December to wear it!
         
        <br />
      </div>
      <div
        style={
          Object {
            "border": "2px solid black",
          }
        }
      >
        Kazza
         
        <br />
        Nov 11, 2020
         
        <br />
        5
         star 
        <br />
        Mask color: 
        Christmas tree+bow
         
        <br />
        Speedy delivery Item nice Recommend
         
        <br />
      </div>
      <div
        style={
          Object {
            "border": "2px solid black",
          }
        }
      >
        Angela Naylor
         
        <br />
        Oct 21, 2020
         
        <br />
        5
         star 
        <br />
        Mask color: 
        Christmas tree+bow
         
        <br />
        Absolutely beautiful will definitely be purchasing some more . Delivery time outstanding so quick thank you so much xx ðŸ’œ
         
        <br />
      </div>
      <div
        style={
          Object {
            "border": "2px solid black",
          }
        }
      >
        ruthiebaby1619
         
        <br />
        Nov 9, 2020
         
        <br />
        5
         star 
        <br />
        Mask color: 
        Christmas tree+bow
         
        <br />
        This mask is fabulous! Glitter design at both sides. Very comfortable. Can't wait till December to wear it!
         
        <br />
      </div>
    </div>
  </div>
</div>
`;

exports[`Component snapshot renders correctly inital layout reviews nav renders 1`] = `
<div>
  Reviews for this item 
  6
   
  <br />
  Review for this shop 1,088
  <div>
    Sort by: Recommended
  </div>
</div>
`;

exports[`Component snapshot renders correctly inital layout single review renders 1`] = `
<div
  style={
    Object {
      "border": "2px solid black",
    }
  }
>
  Kazza
   
  <br />
  Nov 11, 2020
   
  <br />
  5
   star 
  <br />
  Mask color: 
  Christmas tree+bow
   
  <br />
  Speedy delivery Item nice Recommend
   
  <br />
</div>
`;

exports[`Component snapshot renders correctly inital layout total review renders 1`] = `
<div>
  6
   reviews 
  5
   rating stars
</div>
`;
